dask/base.py

```diff
@@ -98,6 +102,48 @@ def _restore_excepthook():
 original_excepthook = sys.excepthook
 sys.excepthook = _clean_traceback_hook(sys.excepthook)
 
+_annotations: ContextVar[dict] = ContextVar("annotations", default={})
+
+
+def get_annotations(default_value=None) -> dict | None:
```

Comment:
Initially, I was going to create a module for this code called `dask.annotations`. Unfortunately, we already have `dask.annotations` here:

https://github.com/dask/dask/blob/a6ee367ab76f4b12ab2dcea49e9b91dff9ba63d0/dask/__init__.py#L1

The name would be clashing, so for now, I left it in `base.py`. If you have better suggestions, let me know.